example/tests/some_module_test.py

```diff
@@ -130,3 +130,22 @@ def test_property():
     # Everything works with properties
     Expect("SomeAPI").to_receive("some_property").and_return("bar")
     assert SomeAPI().some_property == "bar"
+
+
+def test_disable():
```

Comment:
what happens if you run a test right after this one where some mocks are disabled? are the mocks properly re-enabled automatically?

Reply:
To check this out, I just added exact replicas of the tests, but without disabling the Mock. They threw errors saying they required an Expect statement, which means that the mock was re-enabled. 

In `some_module_test.py`, it is obvious why this happens: the `tear_down` method is called in the `run_around_tests` fixture, which internally calls `set_up` again for all the mocked methods, effectivelly resetting them.

In `some_other_module_test.py`, tear_down is also called by the Expectations context manager.

If tear_down is not called in one way or another, then the Mock will remain deactivated, which is why using the context manager is usually the smart thing to do.

Reply:
thanks for elaborating!